src/display/network_utils.js

```diff
@@ -21,6 +21,21 @@ import {
 import { getFilenameFromContentDispositionHeader } from "./content_disposition.js";
 import { isPdfFile } from "./display_utils.js";
 
+function createHeaders(isHttp, httpHeaders) {
+  const headers = new Headers();
+
+  if (!isHttp || !httpHeaders || httpHeaders !== "object") {
```

Comment:
```suggestion
  if (!isHttp || !httpHeaders || typeof httpHeaders !== "object") {
```

Reply:
I'm surprised that tests passed, since this is always ignoring this `httpHeaders` object :sweat_smile:

Reply:
Whoops, thanks for spotting that!

> I'm surprised that tests passed, since this is always ignoring this `httpHeaders` object 😅

The `httpHeaders` API-option was added years ago, specifically for third-party users, however we never used it within the PDF.js project itself and consequently it wasn't covered by tests; so I suppose that in a way it's not that surprising :-)

The patch has been updated with a few (basic) unit-tests for the new helper function.